src/libp2p_multistream_server.erl

```diff
@@ -19,10 +20,20 @@
 start_link(Ref, Connection, Handlers, HandlerOpt) ->
     {ok, proc_lib:spawn_link(?MODULE, init, [{Ref, Connection, Handlers, HandlerOpt}])}.
 
+-spec start_link(libp2p_connection:connection(), [{string(), term()}], any()) -> {ok, pid()}.
+start_link(Connection, Handlers, HandlerOpt) ->
+    {ok, proc_lib:spawn_link(?MODULE, init, [{Connection, Handlers, HandlerOpt}])}.
+
 init({Ref, Connection, Handlers, HandlerOpt}) ->
     ok = libp2p_connection:acknowledge(Connection, Ref),
     self() ! handshake,
-    loop(#state{connection=Connection, handlers=Handlers, handler_opt=HandlerOpt}).
+    TimerRef = erlang:send_after(30000, self(), timeout),
```

Comment:
Make or reuse a constant of that 30000 which would help clarify what this does? Is this a RECV_TIMEOUT?

Reply:
It's the time we allow a session protocol to be negotiated before we kill the connection.